src/puppetlabs/puppetdb/command/dlo.clj

```diff
@@ -1,210 +1,182 @@
-  archive once per `threshold`."
-  [subdir threshold]
-  {:pre [(or (string? subdir)
-             (instance? java.io.File subdir))]}
-  (if-let [archive-time (last-archived subdir)]
-    (after? archive-time (ago threshold))))
-
-(defn- compress-subdir!
-  "Compresses the specified `files` in the particular command-specific
-  subdirectory `subdir`, replacing them with a timestamped .tgz file."
-  [subdir files]
-  {:pre [(or (string? subdir)
-             (instance? java.io.File subdir))]}
-  (create-metrics-for-subdir! subdir)
-  (let [target-file (file subdir (str (kitchensink/timestamp) ".tgz"))
-        temp (str target-file ".partial")]
-    (try
-      (time! (subdir-metric subdir :compression)
-             (apply archive/tar temp "UTF-8" (map (juxt fs/base-name slurp) files))
-             (fs/rename temp target-file)
-             (doseq [filename files]
-               (fs/delete filename)))
-      (catch Throwable e
-        (mark! (global-metric :compression-failures))
-        (mark! (subdir-metric subdir :compression-failures))
-        (throw e))
-      (finally
-        (fs/delete temp)))))
-
-(defn compress!
-  "Compresses all DLO subdirectories which have messages that have been in the
-  directory for longer than `threshold`. This will produce a timestamped .tgz
-  file in each subdirectory containing old messages."
-  [dir threshold]
-  {:pre [(or (string? dir)
-             (instance? java.io.File dir))
-         (period? threshold)]}
-  (create-metrics-for-dlo! dir)
-  (when-let [subdir-files (seq (for [subdir (remove #(already-archived? % threshold) (subdirectories dir))
-                                     :let [files (compressible-files subdir threshold)]
-                                     :when (seq files)]
-                                 [subdir files]))]
-    (time! (global-metric :compression)
-           (doseq [[subdir files] subdir-files]
-             (compress-subdir! subdir files)))))
+  (:require
+   [clojure.java.io :as io]
+   [clojure.string :as str]
+   [metrics.counters :as counters]
+   [puppetlabs.i18n.core :refer [trs]]
+   [puppetlabs.kitchensink.core :refer [timestamp]]
+   [puppetlabs.puppetdb.nio :refer [copts copt-atomic copt-replace oopts]]
+   [puppetlabs.puppetdb.queue :as queue
+    :refer [cmdref->entry metadata-str parse-cmd-filename]]
+   [puppetlabs.stockpile.queue :as stock])
+  (:import
+   [java.nio.file AtomicMoveNotSupportedException
+    FileAlreadyExistsException Files LinkOption]
+   [java.nio.file.attribute FileAttribute]))
+
+(def command-names (cons "unknown" queue/metadata-command-names))
+
+(defn- cmd-counters
+  "Adds gauges to the dlo for the given category (e.g. \"global\"
+  \"replace command\")."
+  [registry category]
+  (let [dns "puppetlabs.puppetdb.dlo"]
+    {:messages (counters/counter registry [dns category "messages"])
+     :filesize (counters/counter registry [dns category "filesize"])}))
+
+(defn- update-metrics
```

Comment:
Hmm, well I'd done this under the assumption that the namespace would be aliased, so you'd have dlo/initialize, etc;  i.e. if we're going to prefer using prefixes (:as), then otherwise you'd have dlo/dlo-initialize, which seems a little redundant.